python/ray/serve/_private/common.py

```diff
@@ -292,3 +292,21 @@ class RunningReplicaInfo:
     actor_handle: ActorHandle
     max_concurrent_queries: int
     is_cross_language: bool = False
+
+    def __post_init__(self):
+        # Set hash value when object is constructed
+        hash_val = hash(
+            " ".join(
+                [
+                    self.deployment_name,
+                    self.replica_tag,
+                    str(self.actor_handle._actor_id),
+                    str(self.max_concurrent_queries),
+                    str(self.is_cross_language),
+                ]
+            )
+        )
+        object.__setattr__(self, "_hash", hash_val)
```

Comment:
For my education what's the benefit of this over `self._hash = hash_val`? If not obvious maybe add a code comment

Reply:
oh it is a hacky way to set attribute, since we use `frozen` for this dataclass class.

Reply:
Makes sense! Would be good to put this in a code comment